cdisc_rules_engine/operations/base_operation.py

```diff
@@ -58,3 +78,138 @@ def _handle_dictionary_result(self, result):
             self.evaluation_dataset
         )
         return self.evaluation_dataset
+
+    def _get_variables_metadata_from_standard(self) -> List[dict]:
+        """
+        Gets variables metadata for the given class and domain from cache.
+        The cache stores CDISC Library metadata.
+
+        Return example:
+        [
+            {
+               "label":"Study Identifier",
+               "name":"STUDYID",
+               "ordinal":"1",
+               "role":"Identifier",
+               ...
+            },
+            {
+               "label":"Domain Abbreviation",
+               "name":"DOMAIN",
+               "ordinal":"2",
+               "role":"Identifier"
+            },
+            ...
+        ]
+        """
+        # get model details from cache
+        cache_key: str = get_standard_details_cache_key(
+            self.params.standard, self.params.standard_version
+        )
+
+        standard_details: dict = self.cache.get(cache_key) or {}
+        model = standard_details.get("_links", {}).get("model")
+        class_details, domain_details = self._get_class_and_domain_metadata(
+            standard_details
+        )
+        model_type, model_version = self._get_model_type_and_version(model)
+        model_cache_key = get_model_details_cache_key(model_type, model_version)
+        model_details = self.cache.get(model_cache_key) or {}
+
+        variables_metadata = domain_details.get("datasetVariables", [])
+        variables_metadata.sort(key=lambda item: item["ordinal"])
+        if class_details.get("name") in DETECTABLE_CLASSES:
+            # if the class is one of Interventions, Findings, Events
+            # -> add General Observation class variables to variables metadata
+            gen_obs_class_metadata: dict = self._get_class_metadata(
+                model_details, GENERAL_OBSERVATIONS_CLASS
+            )
+            identifiers_metadata, timing_metadata = self._group_class_variables_by_role(
+                gen_obs_class_metadata["classVariables"]
+            )
+            # Identifiers are added to the beginning and Timing to the end
+            if identifiers_metadata:
+                identifiers_metadata.sort(key=lambda item: item["ordinal"])
+                variables_metadata = identifiers_metadata + variables_metadata
+            if timing_metadata:
+                timing_metadata.sort(key=lambda item: item["ordinal"])
+                variables_metadata = variables_metadata + timing_metadata
```

Comment:
Mixing ig and class variables is intentional as it was requested by Jozef for getting the allowed IG variables. Is there a case where some variables in the class are not in the model dataset?

Reply:
I mean the case where some variables in the model class are not in the ig domain. I believe those variables are still permissible in the datasets. As for duplicates as of right now there will only be duplicates if their are duplicates in the library between domain and class

Reply:
Additionally the order does not matter in the context of this PBI. If there is a bug in the ordering it should be addressed in another issue

Reply:
> I mean the case where some variables in the model class are not in the ig domain. I believe those variables are still permissible in the datasets. 

Oops, yes, that is what i meant too.

> As for duplicates as of right now there will only be duplicates if their are duplicates in the library between domain and class

In the ig domain, you would have Findings.LB.LBTESTCD for example and in the model class, you would have Findings.--TESTCD
So I think any variable that is in an ig domain would be duplicated here.

> Additionally the order does not matter in the context of this PBI. If there is a bug in the ordering it should be addressed in another issue

This is fine. Then we just need to fix the duplicates issue

Reply:
So is it correct that we should add the class variables to the IG variables, and just remove the duplicates?

Reply:
yes, i think so, but we should check with the team

Reply:
Variable order was discussed pretty thoroughly when writing the operation to get ig column order:

When getting the list of allowed variables for a domain these values should be:
identifier variables + domain variables + timing variables
Identifier and Timing variables only apply when the domain belongs to a general observations class.
Since the list of allowed IG variables does not include class variables other than potentially Identifier and Timing variables the example of duplicates provided should not occur. However, since there could be other duplicates I will handle duplicates when generating that list.

In the case of library model column order - all variables come from the model class anyways so there is not any mixing.

Reply:
Good point - LBTESTCD is a bad example. LBDTC would have been a better example. Sounds good.